src/main/java/service/BookingService.java

```diff
@@ -0,0 +1,70 @@
+package service;
+
+import domain.dao.BookingDao;
+import domain.entity.BookingEntity;
+import domain.entity.FlightEntity;
+
+import java.util.List;
+import java.util.Optional;
+
+public class BookingService {
+
+    private final BookingDao bookingDao;
+    private final FlightService flightService;
+
+    public BookingService(BookingDao bookingDao, FlightService flightService) {
+        this.bookingDao = bookingDao;
+        this.flightService = flightService;
+    }
+
+    public boolean addBooking(BookingEntity bookingEntity) {
+        if (bookingEntity.getFlightId() != null && bookingEntity.getPassengers() != null
+                && !bookingEntity.getPassengers().isEmpty()) {
+            return bookingDao.add(bookingEntity);
+        }
+        return false;
+    }
+
+    public Optional<BookingEntity> getBookingById(String id) {
+        return bookingDao.getById(id);
+    }
+
+    public List<BookingEntity> getAllBookings() {
+        return bookingDao.getAll();
+    }
+
+    public List<BookingEntity> findBookingsByPassengerName(String passengerName) {
+        if (passengerName == null || passengerName.isEmpty()) {
+            throw new IllegalArgumentException("Passenger name cannot be null or empty ");
+        }
+        return bookingDao.findBookingsByPassengerName(passengerName);
+    }
+
+    public boolean cancelBooking(String bookingId) {
+        if (bookingId == null || bookingId.isEmpty()) {
+            throw new IllegalArgumentException("Booking ID cannot be null or empty");
+        }
+        return bookingDao.cancelBooking(bookingId);
+    }
+
+    public boolean updateBooking(BookingEntity bookingEntity) {
+        if (bookingEntity.getId() == null || bookingEntity.getId().isEmpty()) {
+            throw new IllegalArgumentException("Flight ID cannot be null or empty");
+        }
+        return bookingDao.update(bookingEntity);
+    }
+
+    public List<BookingEntity> findBookingsByDestination(String destination) throws IllegalAccessException {
```

Comment:
is documentation has this feature? 

#The passenger is prompted to enter the following information: destination, date, number of people (how many tickets to buy). After that, the program must search for flights that meet the specified conditions (there must be enough seats for all passengers). All found flights should be displayed on the screen. After that, the passenger can choose one of the found flights by specifying its serial number or return to the main menu (selecting item 0). If the passenger decides to book a flight, he will need to enter the names and surnames of all passengers.